package.json

```diff
@@ -106,10 +106,10 @@
     "lint": "standard && tslint -c tslint.json ./src/**/*.ts ./test/**/*.ts",
     "test": "npm run lint && ts-node test | tap-diff",
     "posttest": "rm -rf .tmp",
-    "prerelease": "npm test",
+    "prerelease": "npm run tsc && npm test",
     "release": "sh .scripts/release.sh",
     "tsc": "rm -rf lib && tsc",
-    "prepublish": "npm run tsc"
+    "prepublish": "in-publish && echo 'You need to use \"npm run release\" to publish pnpm' && false || not-in-publish"
```

Comment:
Seems like I can't modify the package.json from a prepublish script, so this implementation is the only way